Source/WebCore/html/HTMLImageElement.cpp

```diff
@@ -658,6 +658,35 @@ String HTMLImageElement::completeURLsInAttributeValue(const URL& base, const Att
     return HTMLElement::completeURLsInAttributeValue(base, attribute, resolveURLs);
 }
 
+Attribute HTMLImageElement::replaceURLsInAttributeValue(const Attribute& attribute, const HashMap<String, String>& replacementURLStrings) const
+{
+    if (attribute.name() != srcsetAttr)
+        return attribute;
+
+    if (replacementURLStrings.isEmpty())
+        return attribute;
+
+    Vector<ImageCandidate> imageCandidates = parseImageCandidatesFromSrcsetAttribute(StringView(attribute.value()));
```

Comment:
How about `picture` element and `source`? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture

Reply:
Also use auto?

Reply:
Updated HTMLSourceElement to extract and replace urls for srcset attribute in latest patch and added test.